boring/src/x509/mod.rs

```diff
@@ -142,6 +206,38 @@ impl X509StoreContextRef {
         }
     }
 
+    pub fn init_without_cleanup(
+        &mut self,
+        trust: store::X509Store,
+        cert: X509,
+        cert_chain: Stack<X509>,
+    ) -> Result<(), ErrorStack> {
+        unsafe {
+            if let Err(e) = cvt(ffi::X509_STORE_CTX_init(
+                self.as_ptr(),
+                trust.as_ptr(),
+                cert.as_ptr(),
+                cert_chain.as_ptr(),
+            )) {
+                ffi::X509_STORE_CTX_cleanup(self.as_ptr());
```

Comment:
the method is called "without cleanup", and still may call cleanup?

Reply:
I'm open to suggestions for a better name btw.

Reply:
Yeah, that works.